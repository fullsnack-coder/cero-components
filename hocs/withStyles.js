import React from 'react'
import { getClasses } from '@glrodasz/storybook-tools-styles'

const withStyles = (styles) => (WrappedComponent) => {
  const WithStylesComponent = (props) => {
    const allProps = { ...WrappedComponent.defaultProps, ...props }
    return (
      <WrappedComponent getStyles={getClasses(styles)(allProps)} {...props} />
    )
  }

  WithStylesComponent.displayName = WrappedComponent.displayName

  return WithStylesComponent
}

export default withStyles
