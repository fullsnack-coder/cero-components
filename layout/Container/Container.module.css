.container {
  display: flex;
  width: 100%;
  max-width: var(--container-max-width);
  height: 100%;
  flex: auto;
  padding: 20px 15px 20px;
  margin: 0 auto;
}

.is-centered {
  justify-content: center;
}

.is-playground {
  border: var(--border-width-thick) dashed var(--color-primary);
}
