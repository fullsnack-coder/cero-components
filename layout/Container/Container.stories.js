import { Container, styles } from '.'

import { getTemplate } from '@glrodasz/storybook-tools-helpers'

const Template = getTemplate(Container, styles)

export default {
  title: 'Layout/Container',
  component: Container,
  args: {
    isPlayground: true,
    children: `Farthings pulled strain? Half they'll greater land sordid Elessar`,
  },
  argTypes: {
    children: { control: 'text' },
  },
}

export const Default = Template.bind({})

export const Centered = Template.bind({})
Centered.args = {
  isCentered: true,
}
