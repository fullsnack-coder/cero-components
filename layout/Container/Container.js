import React from 'react'
import PropTypes from 'prop-types'
import styles from './Container.module.css'
import withStyles from '../../hocs/withStyles'

export const Container = ({
  getStyles,
  children,
  isCentered,
  isPlayground,
}) => {
  return (
    <div
      className={getStyles('container', {
        'is-playground': isPlayground,
        'is-centered': isCentered,
      })}
    >
      {children}
    </div>
  )
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  getStyles: PropTypes.func.isRequired,
  isPlayground: PropTypes.bool,
  isCentered: PropTypes.bool,
}

Container.defaultProps = {
  isPlayground: false,
  isCentered: true,
  getStyles: () => {},
}

export default withStyles(styles)(Container)
