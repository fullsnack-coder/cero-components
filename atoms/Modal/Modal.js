import React, { useState } from 'react'
import PropTypes from 'prop-types'

import styles from './Modal.module.css'
import { options } from './constants'
import withStyles from '../../hocs/withStyles'
import Icon from '../Icon'
import Container from '../../layout/Container'
import isEmpty from '../../utils/isEmpty'

import useMedia from '../../hook/useMedia'

import {
  createHandlerClick,
  createHandlerClose,
  createHandlerSecondaryAction,
} from './handlers'

export const Modal = ({
  onClose,
  secondaryAction,
  children,
  getStyles,
  type,
  isPlayground,
}) => {
  const isDesktop = useMedia(['(min-width: 992px)'], [true])
  const [onFadeOut, setOnFadeOut] = useState(false)

  const handleClose = createHandlerClose({ onClose, setOnFadeOut })

  return (
    <div
      className={getStyles('backdrop', {
        'is-playground': isPlayground,
        'on-fade-out': onFadeOut,
      })}
      onClick={handleClose}
    >
      <div
        className={getStyles('modal', ['type'])}
        onClick={createHandlerClick()}
      >
        <div className={getStyles('heading')}>
          {!!onClose && (
            <Icon
              color={isDesktop && type === 'secondary' ? 'primary' : 'inverted'}
              name={isDesktop ? 'cross' : 'angleLeft'}
              background={isDesktop ? 'transparent' : 'muted'}
              onClick={handleClose}
              size={isDesktop ? 'lg' : 'md'}
            />
          )}
          {secondaryAction && !isEmpty(secondaryAction) && (
            <Icon
              color="inverted"
              name={secondaryAction?.icon}
              background="muted"
              onClick={createHandlerSecondaryAction({
                onSecondaryAction: secondaryAction?.handler,
              })}
            />
          )}
        </div>
        <Container>{children}</Container>
      </div>
    </div>
  )
}

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  getStyles: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  secondaryAction: PropTypes.shape({
    icon: PropTypes.string,
    handler: PropTypes.func,
  }),
  type: PropTypes.oneOf(options.types),
  isPlayground: PropTypes.bool,
}

Modal.defaultProps = {
  getStyles: () => {},
  type: 'primary',
}

export default withStyles(styles)(Modal)
