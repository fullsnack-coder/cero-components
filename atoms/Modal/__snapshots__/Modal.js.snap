// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ storybook ] [ Atoms/Modal ] should render Closable 1`] = `
<DocumentFragment>
  <div
    class="backdrop is-playground"
  >
    <div
      class="modal type-primary"
    >
      <div
        class="heading"
      />
      <div
        class="container"
      >
        Successful boar world's somebody herbs helpful spiders Samwise Gamgee helps unlost circles guard? Investment Thofin folly? Troublemaker minutes moldy Rivendell eve late traveling around crescent fault never naught
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Modal ] should render Default 1`] = `
<DocumentFragment>
  <div
    class="backdrop is-playground"
  >
    <div
      class="modal type-primary"
    >
      <div
        class="heading"
      />
      <div
        class="container"
      >
        Successful boar world's somebody herbs helpful spiders Samwise Gamgee helps unlost circles guard? Investment Thofin folly? Troublemaker minutes moldy Rivendell eve late traveling around crescent fault never naught
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Modal ] should render SecondaryAction 1`] = `
<DocumentFragment>
  <div
    class="backdrop is-playground"
  >
    <div
      class="modal type-primary"
    >
      <div
        class="heading"
      >
        <div
          class="icon color-inverted size-md background-muted is-clickable"
          style="width: 35px; height: 35px;"
        >
          <svg
            height="35"
            viewBox="5 1 20 20"
            width="35"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M14.93 20.86C9.45472 20.86 5 16.4053 5 10.93C5 5.45472 9.45472 1 14.93 1C20.4053 1 24.86 5.45472 24.86 10.93C24.86 16.4053 20.4053 20.86 14.93 20.86ZM14.93 2.09095C10.0559 2.09095 6.09095 6.05591 6.09095 10.93C6.09095 15.8041 10.0559 19.769 14.93 19.769C19.8041 19.769 23.769 15.8041 23.769 10.93C23.769 6.05591 19.8041 2.09095 14.93 2.09095Z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M13.744 15.221L8.67871 10.1423L9.45059 9.37155L13.7458 13.6773L20.41 7.01251L21.1807 7.78439L13.744 15.221Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <div
        class="container"
      >
        Successful boar world's somebody herbs helpful spiders Samwise Gamgee helps unlost circles guard? Investment Thofin folly? Troublemaker minutes moldy Rivendell eve late traveling around crescent fault never naught
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Modal ] should render Types 1`] = `
<DocumentFragment>
  <div
    class="backdrop is-playground"
  >
    <div
      class="modal type-primary"
    >
      <div
        class="heading"
      />
      <div
        class="container"
      >
        Successful boar world's somebody herbs helpful spiders Samwise Gamgee helps unlost circles guard? Investment Thofin folly? Troublemaker minutes moldy Rivendell eve late traveling around crescent fault never naught
      </div>
    </div>
  </div>
  <div
    class="backdrop is-playground"
  >
    <div
      class="modal type-secondary"
    >
      <div
        class="heading"
      />
      <div
        class="container"
      >
        Successful boar world's somebody herbs helpful spiders Samwise Gamgee helps unlost circles guard? Investment Thofin folly? Troublemaker minutes moldy Rivendell eve late traveling around crescent fault never naught
      </div>
    </div>
  </div>
</DocumentFragment>
`;
